src/enrichment_tables/file.rs

```diff
@@ -178,22 +179,20 @@ impl File {
         self.headers.iter().position(|header| header == col)
     }
 
-    /// Currently all matches are case insensitive.
-    /// TODO We want to add a configuration option to allow for case sensitive searches. This will
-    /// allow for more performant matches.
-    fn row_equals(&self, condition: &[Condition], row: &[Value]) -> bool {
+    /// Does the given row match all the conditions specified?
+    fn row_equals(&self, case: Case, condition: &[Condition], row: &[Value]) -> bool {
         condition.iter().all(|condition| match condition {
             Condition::Equals { field, value } => match self.column_index(field) {
                 None => false,
-                Some(idx) => match (&row[idx], value) {
-                    (Value::Bytes(bytes1), Value::Bytes(bytes2)) => {
+                Some(idx) => match (case, &row[idx], value) {
+                    (Case::Insensitive, Value::Bytes(bytes1), Value::Bytes(bytes2)) => {
                         match (std::str::from_utf8(bytes1), std::str::from_utf8(bytes2)) {
                             (Ok(s1), Ok(s2)) => s1.to_lowercase() == s2.to_lowercase(),
                             (Err(_), Err(_)) => bytes1 == bytes2,
                             _ => false,
                         }
                     }
-                    (value1, value2) => value1 == value2,
+                    (_, value1, value2) => value1 == value2,
```

Comment:
Could we use `Case::Sensitive` here? It feels a bit odd to have a catch-all.

Reply:
The pattern is also checking that the other two parameters are `Value::Bytes`. So the catch-all means it doesn't matter what the `Case` is when we don't have `Value::Bytes`.

Reply:
Ah, true, I missed that check for `Value::Bytes`. 👍